src/backend/computeFunction.js

```diff
@@ -0,0 +1,34 @@
+//@flow
+
+import {type WeightedGraph} from "../core/weightedGraph";
+import {TaskReporter} from "../util/taskReporter";
+import {type TimelineCredParameters} from "../analysis/timeline/params";
+import {type PluginDeclaration} from "../analysis/pluginDeclaration";
+import {TimelineCred} from "../analysis/timeline/timelineCred";
+
+/**
+ * An abstract handle for TimelineCred.compute-like functions.
+ */
+export type ComputeFunction = (opts: ComputeOpts) => Promise<TimelineCred>;
+
+// Note: type should not be strict, it's expected to be a subset.
```

Comment:
I don't understand this comment.

Reply:
My bad, Flow calls this "exact" instead of "strict".
https://flow.org/en/docs/types/objects/#toc-exact-object-types
In other words, disallowing extra properties.

In this case we _do_ want extra properties to be allowed.

Reply:
Rephrased the comment.